### PR TITLE
CDPS-955: Add additional role requirement for distinguishing marks image document type

### PR DESCRIPTION
**Pull request by mtac50**
- [x] Added additional role requirement for the distinguishing mark image document type.
- [x] Updated unit test to include the new role requirement.
- [x] Added integration tests for the distinguishing mark image document type.
- [x] Created an abstract document type integration test class to reduce code duplication between the SAR and distinguishing mark document type integration tests.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsdocumentmanagementapi/resource/DistinguishingMarkImageIntTest.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  override val metadata: JsonNode = JacksonUtil.toJsonNode("{ \"prisonNumber\": \"A1234BC\" }")
+  override val serviceName = "distinguishing-marks-api"
+  override val activeCaseLoadId = "KIR"
+  override val username = "TEST_USER"
+  final override val testFileName = "distinguishing-mark-image.png"
+  override val resourcePathOfTestDocument = "test_data/$testFileName"
+  override val documentFileSize: Long = 10465L
+  override val contentType = "image/png"
+  override val bucketName = S3BucketName.DISTINGUISHING_MARK_IMAGES.value
+}
```
Comment (brightonsbox) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsdocumentmanagementapi/resource/DistinguishingMarkImageIntTest.kt:29]: Nice!

File: src/test/resources/test_data/subject-access-request-report-id-4.sql [hunk 2]
```diff
@@ -21,7 +21,7 @@ VALUES
     'subject-access-request-report',
     'pdf',
     21384,
-    'd58e3582afa99040e27b92b13c8f2280',
+    '',
```
Comment (brightonsbox) [inline src/test/resources/test_data/subject-access-request-report-id-4.sql:24]: Why did this need changing?

**Review by brightonsbox — APPROVED** [review]
LGTM

File: src/test/resources/test_data/subject-access-request-report-id-4.sql (hunk 2 above)
Comment (mtac50) [reply src/test/resources/test_data/subject-access-request-report-id-4.sql:24]: The TL;DR is laziness with a hint of logic...

Logic: When documents are added the file hash isn't actually calculated or persisted so the tests were asserting it was returned on a GET and then asserting it was not populated on a POST which seemed illogical. 

Laziness: Removing this meant I could use the same assertion for both the GET and POST document requests.

However now I'm writing this out I feel like I should just pass the expected filehash into the assertion...

**Review by brightonsbox — APPROVED** [review]
LGTM

**Review by joelstobart-moj — APPROVED** [review]
LGTM

**Pull request merged**